vnext/Microsoft.ReactNative/IReactNotificationService.cpp

```diff
@@ -8,77 +8,181 @@
 namespace winrt::Microsoft::ReactNative::implementation {
 
 //=============================================================================
-// ReactNotificationSubscription implementation
+// IReactNotificationSubscription implementation
 //=============================================================================
 
-ReactNotificationSubscription::ReactNotificationSubscription(
-    IReactNotificationSubscription const &parentSubscription,
-    weak_ref<ReactNotificationService> &&notificationService,
-    IReactPropertyName const &notificationName,
-    IReactDispatcher const &dispatcher) noexcept
-    : m_parentSubscription{parentSubscription},
-      m_notificationService{std::move(notificationService)},
-      m_notificationName{notificationName},
-      m_dispatcher{dispatcher} {}
-
-ReactNotificationSubscription::ReactNotificationSubscription(
-    weak_ref<ReactNotificationService> &&notificationService,
-    IReactPropertyName const &notificationName,
-    IReactDispatcher const &dispatcher,
-    ReactNotificationHandler const &handler) noexcept
-    : m_notificationService{std::move(notificationService)},
-      m_notificationName{notificationName},
-      m_dispatcher{dispatcher},
-      m_handler{handler} {}
-
-ReactNotificationSubscription::~ReactNotificationSubscription() noexcept {
-  Unsubscribe();
-}
+// Common interface to share functionality between ReactNotificationSubscription and ReactNotificationSubscriptionView
+MSO_GUID(IReactNotificationSubscriptionPrivate, "09437980-3508-4690-930c-7c310e205e6b")
+struct IReactNotificationSubscriptionPrivate : ::IUnknown {
+  virtual void SetParent(IReactNotificationSubscription const &parentSubscription) noexcept = 0;
+  virtual void CallHandler(IInspectable const &sender, IInspectable const &data) noexcept = 0;
+};
```

Comment:
this seems a little weird? it's a COM interface, but its methods take cppwinrt projection/wrapper objects?

Reply:
This is just an internal private interface. It is inherited from the "classic COM" IUnknown for the lifetime and QI, but otherwise it does not follow any COM conventions or use of other COM things. I do not see yet how I can avoid using it. Though, there must be a way to do it.